course3/homework-unique_ptr/my_unique_ptr.h

```diff
@@ -0,0 +1,93 @@
+#include <cstddef>
+#include <iostream>
+
+template<class T>
+class MyUniquePtr
+{
+public:
+	MyUniquePtr() : mData(nullptr)
+	{
+		std::cout << "MyUniquePtr::ctor" << std::endl;
+	}
+	
+	MyUniquePtr(MyUniquePtr& that)
+	{
+		// Construction one unique pointer from another simply changes the ownership of the data
+		mData = that.mData;
+		that.mData = nullptr;
+		std::cout << "MyUniquePtr::Copy" << std::endl;
+	}
+	
+	MyUniquePtr(T*&& data) : mData(data)
+	{
+		// Constructing from an r-value means we aquire ownership of the data
+		std::cout << "MyUniquePtr::MoveData" << std::endl;
+	}
+
+	~MyUniquePtr()
+	{
+		std::cout << "MyUniquePtr::Destruct" << std::endl;
+
+		if (mData != nullptr)
```

Comment:
You don't need to check it. It is safe to invoke ```delete``` on a null pointer.
http://en.cppreference.com/w/cpp/language/delete